simulator/risc_v/riscv_register/riscv_register.h

```diff
@@ -42,7 +42,11 @@ class RISCVRegister {
     bool is_zero()                 const { return value == RISCV_REG_zero; }
     constexpr bool is_mips_hi()    const { return false; }
     constexpr bool is_mips_lo()    const { return false; }
-    size_t to_rf_index()             const { return value; }
+    static RISCVRegister from_cpu_index( uint8 id) {
+        assert( id < 32u);
+        return RISCVRegister( static_cast<RegNum>( id));
```

Comment:
That ctor should be private now.

Reply:
Or we can even remove it as we did with `MIPSRegister`. I'll do it in a separate PR.